src/shared/components/oncoprint/TooltipUtils.ts

```diff
@@ -219,7 +221,10 @@ export function makeHeatmapTrackTooltip(
                     data_header = 'METHYLATION: ';
                     break;
                 case AlterationTypeConstants.GENERIC_ASSAY:
-                    data_header = 'GENERIC ASSAY: ';
+                    // track for GENERIC_ASSAY type always has the genericAssayType
+                    data_header = `${deriveDisplayTextFromGenericAssayType(
+                        (trackSpec as IHeatmapTrackSpec).genericAssayType!
+                    )}: `;
                     break;
```

Comment:
We can ensure that the track for `GENERIC_ASSAY` will always has genericAssayType since it is validated while importing the data.